src/coordinator/mod.rs

```diff
@@ -409,6 +409,48 @@ impl Coordinator {
         Ok(())
     }
 
+    // Unlike input registers, holding registers are not broadcast by inverters,
+    // but they are interesting nevertheless. Publishing the holding registers
+    // when we connect to an inverter makes it easy for configuration data to be
+    // tracked, which is particularly useful in conjunction with HomeAssistant.
+    async fn inverter_connected(&self, datalog: Serial) -> Result<()> {
+        let inverter = match self.config.enabled_inverter_with_datalog(datalog) {
+            Some(inverter) => inverter,
+            None => bail!("Unknown inverter connected: {}", datalog),
+        };
+
+        if !inverter.publish_holdings_on_connect() {
+            return Ok(());
+        }
+
+        info!("Reading holding registers for inverter {}", datalog);
+
+        // We can only read holding registers in blocks of 40. Provisionally,
+        // there are 6 pages of 40 values.
+        self.read_hold(inverter.clone(), 0_i16, 40).await?;
```

Comment:
I prefer this to the previous for-loop approach. I do wonder if there are inverters that support fewer pages, though, and that if they are, this will stop lxp-bridge from working with them.

I guess we can add configuration or auto-detection if that case shows up though. Does this work with yours @celsworth ?

Reply:
I'll give this a try tomorrow morning :)

Reply:
I suppose if it doesn't work, or there were an abundance of caution, we could make the top register configurable as a stop-gap.